common/include/pcl/point_cloud.h

```diff
@@ -633,6 +635,8 @@ namespace pcl
       {
         points.assign(std::move(ilist));
         width = new_width;
+        if (width == 0)
```

Comment:
Let's move this check to the top (in both places) and test `new_width` instead.

Reply:
Done ;)